handler/src/main/java/io/netty/handler/ssl/SslContext.java

```diff
@@ -673,6 +673,204 @@ static ApplicationProtocolConfig toApplicationProtocolConfig(Iterable<String> ne
         return apn;
     }
 
+    /**
+     * Creates a builder for new client-side {@link SslContext}.
+     */
+    public static Builder newBuilderForClient() {
+        return new Builder(true);
+    }
+
+    /**
+     * Creates a builder for new server-side {@link SslContext}.
+     *
+     * @param keyCertChainFile an X.509 certificate chain file in PEM format
+     * @param keyFile a PKCS#8 private key file in PEM format
+     */
+    public static Builder newBuilderForServer(File keyCertChainFile, File keyFile) {
+        return new Builder(false).keyManager(keyCertChainFile, keyFile);
+    }
+
+    /**
+     * Creates a builder for new server-side {@link SslContext}.
+     *
+     * @param keyCertChainFile an X.509 certificate chain file in PEM format
+     * @param keyFile a PKCS#8 private key file in PEM format
+     * @param keyPassword the password of the {@code keyFile}, or {@code null} if it's not
+     *     password-protected
+     */
+    public static Builder newBuilderForServer(File keyCertChainFile, File keyFile,
+            String keyPassword) {
+        return new Builder(false).keyManager(keyCertChainFile, keyFile, keyPassword);
+    }
+
+    /** Builder for configuring a new SslContext for creation. */
+    public static final class Builder {
```

Comment:
> I wasn't referring to optional things like trust manager and mutual auth; those are simply specified on the builder with normal methods after builder construction.

Understood.  I was referring to the server static methods which create a server Builder, but that may also be applicable for the client.  I think having them for just server or client is confusing when they may be applicable for both modes.  I think it will simplify things a bit to choose 1 method (builder or static factory methods). The builder pattern is pretty mainstream and IDEs should make it pretty straight forward to figure out what constructors/methods do.